### PR TITLE
centralized form validator

### LINKED ISSUE #27: Clean up auth.tsx
- [x] Put form validation function in separate file

### PR DESCRIPTION
**Pull request by devalparikh**
moved function to its own file

### CODE REVIEW

File: frontend/src/components/form/formValidator.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+            return (<p className="error-text">{input_name} is required</p>)
+        case "pattern":
+            return (<p className="error-text">{input_name} needs to be a valid email address</p>)
+        case "minLength":
+            return (<p className="error-text">{input_name} needs to be greater than 3 characters and less than 20 characters.</p>)
+        case "validate":
+            return (<p className="error-text"> {input_name} needs to match</p>)
+        case "backend":
+            return (<p className="error-text"> {input_name} </p>)
+
```
Comment (devalparikh) [inline frontend/src/components/form/formValidator.tsx:16]: Put todo for other cases

Comment (devalparikh) [reply frontend/src/components/form/formValidator.tsx:16]: fixed

File: frontend/src/components/cards/authCard/AuthCard.tsx [hunk 2]
```diff
@@ -55,28 +56,6 @@ export default function AuthCard(props: iAuthCard) {
     }
 
 
-    // TODO: Make this into a react hook
-    const handleFormValidation = (input_name: string, error_type: string) => {
-        // console.log("error msg: " + input_name + " " + "error type:" + error_type);
```
Comment (jeromesteve202) [inline frontend/src/components/cards/authCard/AuthCard.tsx:60]: Why is this comment still here?

**Review by alexandrofuentelzas — APPROVED** [review]
Great job sir

**Review by jeromesteve202 — COMMENTED** [review]
check it out.

File: frontend/src/components/cards/authCard/AuthCard.tsx (hunk 2 above)
Comment (devalparikh) [reply frontend/src/components/cards/authCard/AuthCard.tsx:60]: to uncomment later for console logging inputs

**Review by jeromesteve202 — APPROVED** [review]
Good job.

**Pull request merged**